src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt

```diff
@@ -291,16 +291,22 @@ class PostgresLinkingQueryService(private val hds: HikariDataSource, private val
         }
     }
 
-    override fun getEntityKeyIdsOfLinkingIds(linkingIds: Set<UUID>): PostgresIterable<Pair<UUID, Set<UUID>>> {
+    override fun getEntityKeyIdsOfLinkingIds(
+            linkingIds: Set<UUID>,
+            normalEntitySetIds: Set<UUID>?
+    ): PostgresIterable<Pair<UUID, Set<UUID>>> {
         return PostgresIterable(
                 Supplier {
                     val connection = hds.connection
-                    val ps = connection.prepareStatement(ENTITY_KEY_IDS_OF_LINKING_IDS_SQL)
+                    val ps = connection.prepareStatement(buildEntityKeyIdsOfLinkingIdsSql(normalEntitySetIds != null))
                     val linkingIdsArray = PostgresArrays.createUuidArray(connection, linkingIds)
 
                     val partitions = PostgresArrays.createIntArray( connection, getPartitionsInfo( linkingIds, partitionManager.getAllPartitions() ))
```

Comment:
This should be probably a separate ticket (if it's not already existing) but since we changed partitions for linked entities to be mapped by their normal entity key ids, this won't give any results.
In case, when entity set ids are non-null, I think it's safe to use all partitions for those entity sets (since the actual partitions of the linking entities will be a subset of those)

Reply:
Btw I couldn't find a ticket for this

Reply:
now there is one https://jira.openlattice.com/browse/LATTICE-2337